### PR TITLE
Add extension method to get standard SLVS user folder

### PR DESCRIPTION
**Pull request by duncanp-sonar**
- addIEnvironmentVariableProvider.GetFolderPath
- refactored existing code

Prep for #4170

### CODE REVIEW

**Comment by duncanp-sonar** [comment]
One of the options for the new settings location is in a folder under `{user}\{appdata}`.
We already have several places that write files under `{user}\{appdata}\SonarLint for Visual Studio`.
Unfortunately, there are two problems with the existing code:
1) we don't have an abstraction over `Environment.GetFolderPath` for testing, and 
2) the root folder path isn't calculated in a single place - the code is duplicated.

The PR cleans up the existing code by:
1) adding a `GetFolderPath` wrapper to `IEnvironmentVariableProvider` and
2) adding an extension method `GetSLVSAppDataRootPath` that returns the standard ``SonarLint for Visual Studio` folder.

I've fixed up the existing code to use the new extension method.

File: src/Core.UnitTests/EnvironmentVariableProviderExtensionsTests.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace SonarLint.VisualStudio.Core.UnitTests
+{
+    [TestClass]
+    public class EnvironmentVariableProviderExtensionsTests
+    {
+        [TestMethod]
+        public void GetSLVSAppDataRootPath_ExpectedValue()
```
Comment (duncanp-sonar) [inline src/Core.UnitTests/EnvironmentVariableProviderExtensionsTests.cs:32]: Test for the new extension method that returns the standard "SonarLint for Visual Studio" folder.

File: src/Core.UnitTests/Notifications/DisabledNotificationsStorageTests.cs [hunk 2]
```diff
@@ -36,7 +36,10 @@ namespace SonarLint.VisualStudio.Core.UnitTests.Notifications
     [TestClass]
     public class DisabledNotificationsStorageTests
     {
-        private string ExpectedDisabledNotificationsFilePath => GetExpectedDisabledNotificationsFilePath();
+        private const string DefaultRootPath = "c:\\users\\someuser";
```
Comment (duncanp-sonar) [inline src/Core.UnitTests/Notifications/DisabledNotificationsStorageTests.cs:39]: Fixing up the existing tests so they pass. The existing tests are a bit messy, but refactoring them was beyond the scope of this commit.

File: src/Core.UnitTests/Notifications/DisabledNotificationsStorageTests.cs [hunk 3]
```diff
@@ -36,7 +36,10 @@ namespace SonarLint.VisualStudio.Core.UnitTests.Notifications
...
+        private const string HardCodedVSVersion = "17";
```
Comment (duncanp-sonar) [inline src/Core.UnitTests/Notifications/DisabledNotificationsStorageTests.cs:40]: Miscellaneous: `17` was a magic number that was used in a couple of separate-but-related methods. I've made it a constant to make the connection between the locations in code a little bit more obvious.

File: src/Integration.Vsix/Analysis/UserSettingsProvider.cs [hunk 4]
```diff
@@ -33,7 +33,7 @@ internal sealed class UserSettingsProvider : IUserSettingsProvider, IDisposable
         // Note: the data is stored in the roaming profile so it will be sync across machines
         // for domain-joined users.
         public static readonly string UserSettingsFilePath = Path.GetFullPath(
-            Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "SonarLint for Visual Studio", "settings.json"));
+            Path.Combine(EnvironmentVariableProvider.Instance.GetSLVSAppDataRootPath(), "settings.json"));
```
Comment (duncanp-sonar) [inline src/Integration.Vsix/Analysis/UserSettingsProvider.cs:36]: I've changed this property to use the new extension method and manually tested the result.
Refactoring the code further was out of the scope of this commit (by this point, I was beginning to wish I had just added the new wrapper and extension method, and created a separate ticket to handle refactoring the existing code at some point in the future).

**Review by bigfluffycookie — APPROVED** [review]
lgtm

**Pull request merged**